Skrell can hold stuff in their tentacles

Tayyyyyyy opened the pull request:
Implements Phantasmic Dream's idea from [here](https://nanotrasen.se/forum/topic/9859-skrell-headpockets/), since Skrells need some love.

![headpocket_screen](https://cloud.githubusercontent.com/assets/26497062/26387628/edc45508-4003-11e7-80d7-5ee79f876d7d.png)

The goal was to make this more of a fun little convenience rather than a storage implant lite to avoid powergaming.

- Activating the ability opens up the pocket for putting stuff in
- People close to you can see you put stuff in, as with a regular container.
- Container size can be changed, though right now it's set to hold only one pocket sized item
- Not having tentacles (bald Skrells) will cause stuff to fall out immediately after you put it in
- Being stunned will cause you to lose control of your tentacles and the items to fall out onto the ground. This will incentivize antags to not put important stuff in their headpocket (to avoid affecting balance). Dying or having the pocket surgically removed will also do this.
- Headpocket can be stripped by someone else through the normal strip panel (scroll down for screenshot)
- Works properly even if you have a storage implant or after surgically taking it out and putting it back in
- Skrells cannot have their tentacles shaved off anymore
- EMPs will still work through the tentacles and affect items in the headpocket
- PAIs or whatever else small enough to fit inside can hear stuff outside (and unfold their chassis)
- Fixed a minor grammar issue with the message for when stuff is too big to fit in a container

Concerns:
- ~~Not sure if this is OP or not~~ After further testing, I think Skrell alcohol weakness ensures that this doesn't make Skrell human+. 15u of ethanol will instantly make a Skrell start slurring, then about 5 seconds later cause the Skrell to no longer be able to walk properly. About 15 seconds later, the Skrell falls down and becomes stunned for about 1:40, which is stronger than neuro-toxin, I think. For reference, a human only experiences slurring for a little before returning to normal with the same ethanol dosage.
- icon could probably be better, but I think the new one looks a little better than the original
- ~~Currently stuff in here doesn't count for theft objectives. Will fix.~~ Turns out the existing code is robust enough to detect stuff inside the headpocket! Yay!
![headpocket_theft](https://cloud.githubusercontent.com/assets/26497062/26381480/03c59012-3fda-11e7-8401-d94c3470073a.png)

Thoughts?

:cl: Tayyyyyyy, PhantasmicDream
add: Skrell can put a pocket sized item in their head tentacles, which can be dislodged via stunning, stripping, or death
tweak: Skrell can no longer have their tentacles shaved off
fix: Minor grammar fix for putting stuff that's too big into containers
/:cl:

Xantholne commented:
:erp:

Tayyyyyyy commented on code/modules/surgery/organs/subtypes/skrell.dm:
```diff
@@ -1,3 +1,51 @@
+
+/obj/item/organ/internal/headpocket/ui_action_click()
+	if(!loc)
+		loc = owner
```
Hopefully this doesn't break anything? Internal organs are allowed to have a loc right? If not, I can just set loc back to null after the MouseDrop call.

SamHPurp commented:
Why not. Feels alien, we need more things that feel alien.

Birdtalon commented:
My only concern here is that there really should be a way of accessing this storage for another player that does not require hitting the Skrell.

Tayyyyyyy commented:
@Birdtalon Good point. I'm not quite sure how to interface into the whole drag-to-your-character interface that pops up, but that's definitely something I'll try to fix if people like this idea

Anoonki commented:
Eight damage to the head can be done by throwing something and having it hit you, it's really nothing serious. Maybe have it fall out on being stunned in general? As I assume the reason it falls out when you die is because it takes a little bit of focus to keep it there?

Tayyyyyyy commented:
Yeah earlier I was thinking it'd be annoying to have stuff fall out every time you get stunned but that's probably preferable to having to be hit. And it'll incentivize you not to hide important stuff in there. If it's possible to add a button to the search interface I'll just use that instead, but if not, I'm thinking changing to stunning is probably a good idea.

Ralta commented:
Cool idea - as Sam said, it feels alien which is good. Doesn't seem particularly OP to me, on the basis that it can only hold small things, and isn't usable like a hand so it's a fairly minor buff (albeit one that makes me wanna try Skrell). Long as you can make it searchable, I'm happy!
+1

Fruerlund commented:
Very much fan of this. Species diversity/alien ish. 
The only issue I see at the moment, is the lack of the ability for others to remove the object without being physical. However I can see that being solved

tigercat2000 commented:
Skrell are literally tentacle monsters now. I love it.

tigercat2000 commented on code/game/objects/items/weapons/cosmetics.dm:
```diff
@@ -133,7 +133,7 @@
 				to_chat(user, "<span class='warning'>You find yourself disappointed at the appalling lack of hair.</span>")
 				return
-			if(C.h_style == "Bald" || C.h_style == "Balding Hair" || C.h_style == "Skinhead")
+			if(C.h_style == "Bald" || C.h_style == "Balding Hair" || C.h_style == "Skinhead" || findtextEx(C.h_style, "Skrell"))
```
It would probably be better to just add another check to this like the above
```
if(C.get_species() == "Skrell")
    to_chat(user, "<span class='warning'>Your razor isn't going to cut through tentacles.</span>")
    return
```

KasparoVy commented:
Shouldn't putting sharp objects in the head pocket be bad? 'Cuz flesh?

taukausanake commented:
What about when a Skrell rests or falls asleep? Do those count as a type of stun and will they drop the thing from their head? And what do you mean by surgical? Can you transplant head tentacles on to other mobs? Because that sounds awesome.

Sure would be nice to have this also implemented for Wyrn and their stubby extra arms. Always thought those would make a nice pair of pocket spots as lore states those arms are not used for much else

Fox-McCloud commented:
This makes them objectively superior to humans as a playable race, of which makes me highly skeptical, instantly.

sothangel commented:
A nice mechanical difference that makes aliens seem more alien. I'm happy with it.

SamHPurp commented:
@Fox-McCloud And vulps aren't?

Anoonki commented:
Spaceproofing < A pocket slot that you drop items from if you get punched in the nose

This just in.

Spacemanspark commented:
@Fox-mccloud I don't really see how. Regardless, Skrell needs some love anyways. :+1:

TullyBurnalot commented:
@SamHPurp @Anoonki Stay on point and snark levels down, please, we can discuss this without opening the door to the same-old same-old.

SamHPurp commented:
@TullyBurnalot no snark intended. There are several races that are objectively better than humans at certain things.

My argument is that that isn't a bad thing.

Tayyyyyyy commented:
@taukausanake The codebase makes a distinction between resting/sleeping and stunning, so resting or going SSD or something won't cause it to dislodge. I just tested it and yes, you can transplant the headpocket from a skrell to a different species. However, if you don't have tentacles, stuff will just fall right out of it, so basically this is limited to only skrell.

@Fox-McCloud I was thinking this would help balance out the inability to process alcohol. Being able to breathe underwater kind of isn't used often. The intention is to add a bit of flavor to skrell without buffing them too much.

@KasparoVy I think it would be possible to override the internal storage and inspect what's being stored. Figuring out what's sharp, however, might be a different matter...

TullyBurnalot commented:
Considering what can be put in pockets, this could be a nice utility addition that won't really impact much.

PhantasmicDream commented:
Ohmygosh my idea. 
I'd hug you if I could!

FlattestGuitar commented:
I like it. The change is small enough that it really doesn't matter if they're superior to humans. The amount of impact on gameplay this will have is tiny, and I feel it help make our species more diverse.

I have no concern outside of what people have already said here.

:+1: x 💯

HereticChurch commented:
> Skrells cannot have their tentacles shaved off anymore

Amazing. 
The rest of the PR is great too.

Tayyyyyyy commented:
Thanks for all the feedback everyone <3
I've made it so you can use the strip panel to dislodge the headpocket now. Kept the stun mechanic cause I think that'll keep people from powergaming the pocket. Sorry for the small buttons...such is playing ss13 on a high dpi screen.
![dislodging](https://cloud.githubusercontent.com/assets/26497062/26381162/8d657330-3fd7-11e7-9363-f483d54acc24.png)

SamHPurp commented:
@Tayyyyyyy thats for making that change, I think it silenced my only remaining qualm!

👍

scrubmcnoob commented:
Okay the idea is neat but this objectively makes them better to humans.
We once had this problem with vulps and taj as they got darksight with no inherent disabilities but this was amended with colorblindness. Whether you agree the nerf is proportional to the buff is up for another discussion, the point is that there was a direct nerf because of their benefit for darksight.

So what will the nerf be to compensate for this extra pocket. We don't want a darksight balance pr fiasco part two skrell edition.

taukausanake commented:
@Tayyyyyyy Make Skrell head tentacles into an organ, maybe in a later PR. Transplanting the pocket is possible you say but does nothing if you don't have tentacles so you should transplant the pocket with their carrier.

And only because I'm the only one who would think this, but the head tentacles lore-wise are where Skrell eggs are fertilized and are moist for storing eggs. Someone sticking their hand in there might be a little... awkward? Food for thought

HereticChurch commented:
@scrubmcnoob Skrell already have an innate weakness to alcohol, which could be said to be a harsher downside than the colourblindness.

scrubmcnoob commented:
If they already got a nerf with no compensation, then I don't see a problem with this.

Tayyyyyyy commented:
@taukasnake that's probably the ideal implementation, though I'm not sure how complex doing it that way would be. Regarding that bit of lore, I guess Skrells now have yet another reason to tell shitcurity :P 

@scrubmcnoob to be 100% clear, Skrell are able to breathe underwater. However I think that rarely comes into play; drowning by pool is probably the least likely way for you to die. For that reason, I think Skrells just end up being really underplayed, which is unfortunate cause lore wise, Skrell and humans are basically besties since they met each other first and have their alliance and everything.

taukausanake commented:
@Tayyyyyyy What if hair was an organ? I mean that's what Skrell tentacles are at least, their "hair". Maybe you can scalp people and transplant it on someone else. Extreme barbering. Then you can give people slimepeople hair

HereticChurch commented:
That seems a bit outside the scope of this PR.

taukausanake commented:
Just throwing out suggestions for future

Tayyyyyyy commented:
A refactor of that magnitude might also be outside the scope of my skill level :P Sounds like a good idea, though.

Tayyyyyyy commented:
Okay, just fixed a last couple of problems I thought of. I've fixed the headpocket shielding stuff from EMPs, so EMPs will affect whatever's inside. PAIs can now hear stuff from inside the pocket.

Can't think of any other edge cases I'm missing, so I'm gonna take off the [wip] tag; I think this is ready for an in-depth review. Thanks all for the help!!

Anoonki commented:
@TullyBurnalot My bad, just really really crispy about Vox.

taukausanake commented:
Mmmmm, crispy Vox. Seasoned Vox roasts when?

Spacemanspark commented:
Kentucky fried voxxy

SamHPurp commented:
The secret herbs and spices is just salt

Necaladun commented:
Really love how unique and different this is rather than just x% more or less damage/whatever. Gives them a real alien feel.

Tayyyyyyy commented on code/modules/surgery/organs/subtypes/skrell.dm:
```diff
@@ -1,3 +1,51 @@
+	actions_types = list(/datum/action/item_action/organ_action/toggle)
+	var/obj/item/weapon/storage/internal/pocket
+
+/obj/item/organ/internal/headpocket/New()
```
I tried removing the New()...turns out, can't be done because src and owner aren't defined until the constructor is called :/

pinatacolada replied:
you shouldn't be needing those, have you tried adding
`storage_slots = 1`
`max_w_class = WEIGHT_CLASS_TINY`
`max_combined_w_class = WEIGHT_CLASS_TINY`
to the declaration? Everything else isn't needed, and you can get rid of New() this way

pinatacolada replied:
oh waaiiit i think i get the problem and i'm also blind

SamHPurp commented:
If someone is decapitated, does this remain in the pocket?

Tayyyyyyy commented:
It releases on death, so decapitation or nukies killing the captain shouldn't be an issue, I think...

pinatacolada commented:
point still stands, you should have a way to remove things from it without needing to transplanting it into a skrell

Tayyyyyyy commented:
Ohh I see what you're talking about now. Sorry, didn't understand that piece of code you linked earlier. I already have it release on removal, but maybe I should make it so that it releases on punching it or something just to be safe?

granodd commented:
Having it release on death and/or stun + removal of the tentacles sounds like enough as is. If everything in the tentacle slot popped out upon being hit then there'd be no practical reason to put anything up there unless you were going to discard it anyways.

pinatacolada commented:
Oh I missed the onremoval(). you can make an attackby() to remove it, but I think it's only worth it if you can still put things in it when it's removed, considering it's a storage item disguised as an organ.

Tayyyyyyy commented:
Okay, in that case, I'll leave it as is :P I haven't overridden the organ in enough ways to allow it to operate outside of having the ability in your skills, so you can't actually put anything in it when it's outside. The image of carrying a piece of Skrell around with you to store an item is pretty funnily terrifying though.

Hate9 commented:
Can we please have an actual tentacle icon, and not that horrible pocket icon?

Tayyyyyyy commented:
I don't think a tentacle icon would be a good idea, since removing the organ would produce a tentacle, which might be a little confusing. Willing to use a different icon though if you'd like to make one. I haven't quite got the art style of the game down.

taukausanake commented:
The tentacle is technically the pocket, that is the soft slimy underside of it. Which was why I mentioned transplant of the tentacles themselves. Can't have the pocket without Skrell hair/tentacles.

Again, hair organs so you can scalp people when?

tigercat2000 commented:
@Tayyyyyyy Do you plan to do any further changes to this PR?

Tayyyyyyy commented:
@tigercat2000 Nope, unless someone offers up a different icon or discovers a bug!
